pkg/operator/ceph/cluster/osd/topology/topology.go

```diff
@@ -148,3 +148,64 @@ func GetDefaultTopologyLabels() string {
 
 	return strings.Join(Labels, ",")
 }
+
+// CheckTopologyConflicts verifies that:
+// 1. No child domain (e.g. rack) has fewer distinct values than its parent (e.g. zone).
+// 2. No topology value is used under more than one label key.
+func CheckTopologyConflicts(nodes []corev1.Node) error {
```

Comment:
This is minor, but I would suggest using a pointer to the node list here. There could be clusters with hundreds of nodes, and those could get fairly large on the stack. This is a simple way to save an unknown (but possibly large-ish) amount of RAM.

```suggestion
func CheckTopologyConflicts(nodes *[]corev1.Node) error {
```